pyBumpHunter/bumphunter_2dim.py

```diff
@@ -719,401 +781,490 @@ def SignalInject(self,sig,bkg,is_hist=False):
+            global_inf = Sinf / self.Npe
+            global_sup = Ssup / self.Npe
+            print(
+                "Global p-value : {:1.4f}  ({} / {})   {:1.4f}  ({})   {:1.4f}  ({})".format(
+                    self.global_Pval, S, self.Npe, global_inf, Sinf, global_sup, Ssup
+                )
+            )
+
             # If global p-value is exactly 0, we might have trouble with the significance
-            if(self.global_Pval<0.0000000000000001):
-                self.significance = norm.ppf(1-0.0000000000000001)
+            if self.global_Pval < 0.0000000000000001:
+                self.significance = norm.ppf(1 - 0.0000000000000001)
             else:
-                self.significance = norm.ppf(1-self.global_Pval)
-            
-            if(global_inf<0.0000000000000001):
-                sigma_inf = norm.ppf(1-0.0000000000000001)
+                self.significance = norm.ppf(1 - self.global_Pval)
+
+            if global_inf < 0.0000000000000001:
+                sigma_inf = norm.ppf(1 - 0.0000000000000001)
             else:
-                sigma_inf = norm.ppf(1-global_inf)
-            
-            if(global_sup<0.0000000000000001):
-                sigma_sup = norm.ppf(1-0.0000000000000001)
+                sigma_inf = norm.ppf(1 - global_inf)
+
+            if global_sup < 0.0000000000000001:
+                sigma_sup = norm.ppf(1 - 0.0000000000000001)
             else:
-                sigma_sup = norm.ppf(1-global_sup)
-            print('Significance = {0:1.5f} ({1:1.5f}  {2:1.5f})'.format(self.significance,sigma_inf,sigma_sup))
-            print('')
-            
+                sigma_sup = norm.ppf(1 - global_sup)
+            print(
+                "Significance = {:1.5f} ({:1.5f}  {:1.5f})".format(
+                    self.significance, sigma_inf, sigma_sup
+                )
+            )
+            print("")
+
             # Append reached significance to sigma_ar (with sup and inf variations)
-            self.sigma_ar.append([self.significance,abs(self.significance-sigma_inf),abs(self.significance-sigma_sup)])    
-        
+            self.sigma_ar.append(
+                [
+                    self.significance,
+                    abs(self.significance - sigma_inf),
+                    abs(self.significance - sigma_sup),
+                ]
+            )
+
         # End of injection loop
-        print('REACHED SIGMA LIMIT')
-        print('   Number of signal event injected : {}'.format(self.signal_min))
-        
+        print("REACHED SIGMA LIMIT")
+        print(f"   Number of signal event injected : {self.signal_min}")
+
         # Compute signal strength
-        self.signal_ratio = abs(self.signal_min/self.signal_exp)
-        print('   Signal strength : {0:1.4f}'.format(self.signal_ratio))
-        print('')
-        
+        self.signal_ratio = abs(self.signal_min / self.signal_exp)
+        print(f"   Signal strength : {self.signal_ratio:1.4f}")
```

Comment:
Use f-strings wherever possible

Reply:
They are cleaner, and actually faster too. :) pyupgrade can convert some of them for you. (though, this used to be 3.5+, so that wouldn't have been legal before)